src/lib/ffi/ffi.cpp

```diff
@@ -1699,6 +1757,64 @@ int botan_pubkey_ed25519_get_pubkey(botan_pubkey_t key,
 #endif
    }
 
+int botan_pubkey_load_ecdsa(botan_pubkey_t* key,
+                                   const botan_mp_t public_x,
+                                   const botan_mp_t public_y,
+                                   const char* curve_name)
+   {
+
+   std::unique_ptr<Botan::ECDSA_PublicKey> p_key;
+   if(!pubkey_load_ec(p_key, safe_get(public_x), safe_get(public_y), curve_name))
+      {
+      *key = new botan_pubkey_struct(p_key.release());
+      return 0;
+      }
+
+   return -1;
+   }
+
+int botan_pubkey_load_ecdh(botan_pubkey_t* key,
+                                   const botan_mp_t public_x,
+                                   const botan_mp_t public_y,
+                                   const char* curve_name)
+   {
+   std::unique_ptr<Botan::ECDH_PublicKey> p_key;
+   if(!pubkey_load_ec(p_key, safe_get(public_x), safe_get(public_y), curve_name))
+      {
+      *key = new botan_pubkey_struct(p_key.release());
+      return 0;
+      }
+
+   return -1;
+   }
+
+int botan_privkey_load_ecdsa(botan_privkey_t* key,
+                          const botan_mp_t scalar,
+                          const char* curve_name)
+   {
+   std::unique_ptr<Botan::ECDSA_PrivateKey> p_key;
+   if(!privkey_load_ec(p_key, safe_get(scalar), curve_name))
```

Comment:
If the user passes something that isn't a `botan_mp_t` for the scalar, `safe_get` will throw. Instead pass the `botan_mp_t` to the helper and dereference it in the `try`/`catch` block